great_expectations/agent/actions/run_onboarding_data_assistant.py

```diff
@@ -0,0 +1,94 @@
+        try:
+            self._context.get_expectation_suite(
+                expectation_suite_name=expectation_suite_name
+            )
+            raise ValueError(
+                f"Onboarding Assistant Expectation Suite `{expectation_suite_name}` already exists. "
+                + "Please rename or delete suite and try again"
+            )
+        except StoreBackendError:
+            # resource is unique
+            pass
+
+        try:
+            self._context.get_checkpoint(name=checkpoint_name)
+            raise ValueError(
+                f"Onboarding Assistant Checkpoint `{checkpoint_name}` already exists. "
+                + "Please rename or delete Checkpoint and try again"
+            )
+        except StoreBackendError:
+            # resource is unique
+            pass
+
+        datasource = self._context.get_datasource(datasource_name=event.datasource_name)
+        if not isinstance(datasource, FluentDatasource):
+            raise ValueError(
+                "The RunOnboardingDataAssistant Action can only be used with a fluent-style datasource."
+            )
+        asset = datasource.get_asset(asset_name=event.data_asset_name)
+        try:
+            batch_request = asset.build_batch_request()
+        except ValueError as e:
+            raise ValueError(
+                "The RunOnboardingDataAssistant Action cannot be used with an in-memory dataframe asset."
+            ) from e
+
+        data_assistant_result = self._context.assistants.onboarding.run(
+            batch_request=batch_request,
+        )
+        expectation_suite: ExpectationSuite = (
+            data_assistant_result.get_expectation_suite(
+                expectation_suite_name=expectation_suite_name
+            )
+        )
+        self._context.add_expectation_suite(expectation_suite=expectation_suite)
+
+        checkpoint_config = {
+            "name": checkpoint_name,
+            "validations": [
+                {
+                    "expectation_suite_name": expectation_suite_name,
+                    "expectation_suite_ge_cloud_id": expectation_suite.ge_cloud_id,
+                    "batch_request": {
+                        "datasource_name": event.datasource_name,
+                        "data_asset_name": event.data_asset_name,
+                    },
+                }
+            ],
+            "config_version": 1,
+            "class_name": "Checkpoint",
+        }
+
+        checkpoint = self._context.add_checkpoint(**checkpoint_config)  # type: ignore[arg-type]
+
+        return ActionResult(
+            id=id,
+            type=event.type,
+            created_resources=[
+                CreatedResource(
+                    id=expectation_suite.ge_cloud_id, type="ExpectationSuite"  # type: ignore[arg-type]
+                ),
+                CreatedResource(id=checkpoint.ge_cloud_id, type="Checkpoint"),  # type: ignore[arg-type]
```

Comment:
Please add a comment to these explaining what mypy error is.
Or just copy and paste the exact warning.
Either it's being used incorrectly or we need to update some function signatures.

Reply:
`ge_cloud_id` is typed as an optional string:
```
great_expectations/agent/actions/run_onboarding_data_assistant.py:90: error: Argument "id" to "CreatedResource" has incompatible type "Optional[str]"; expected "str"  [arg-type]
great_expectations/agent/actions/run_onboarding_data_assistant.py:92: error: Argument "id" to "CreatedResource" has incompatible type "Optional[str]"; expected "str"  [arg-type]
```
but since we just received it from cloud, we know it's a valid ID

Reply:
well, we received it from cloud so we assume its a valid ID.. i'll add a check to make that explicit

Reply:
the checkpoint usage is taken from the docs, and creates a number of mypy errors:
```
great_expectations/agent/actions/run_onboarding_data_assistant.py:83: error: Argument 1 to "add_checkpoint" of "CloudDataContext" has incompatible type "**Dict[str, object]"; expected "Optional[str]"  [arg-type]
great_expectations/agent/actions/run_onboarding_data_assistant.py:83: error: Argument 1 to "add_checkpoint" of "CloudDataContext" has incompatible type "**Dict[str, object]"; expected "Union[int, float]"  [arg-type]
great_expectations/agent/actions/run_onboarding_data_assistant.py:83: error: Argument 1 to "add_checkpoint" of "CloudDataContext" has incompatible type "**Dict[str, object]"; expected "str"  [arg-type]
great_expectations/agent/actions/run_onboarding_data_assistant.py:83: error: Argument 1 to "add_checkpoint" of "CloudDataContext" has incompatible type "**Dict[str, object]"; expected "Optional[Dict[Any, Any]]"  [arg-type]
great_expectations/agent/actions/run_onboarding_data_assistant.py:83: error: Argument 1 to "add_checkpoint" of "CloudDataContext" has incompatible type "**Dict[str, object]"; expected "Optional[Sequence[ActionDict]]"  [arg-type]
great_expectations/agent/actions/run_onboarding_data_assistant.py:83: error: Argument 1 to "add_checkpoint" of "CloudDataContext" has incompatible type "**Dict[str, object]"; expected "Optional[List[Dict[Any, Any]]]"  [arg-type]
great_expectations/agent/actions/run_onboarding_data_assistant.py:83: error: Argument 1 to "add_checkpoint" of "CloudDataContext" has incompatible type "**Dict[str, object]"; expected "Union[str, List[str], None]"  [arg-type]
great_expectations/agent/actions/run_onboarding_data_assistant.py:83: error: Argument 1 to "add_checkpoint" of "CloudDataContext" has incompatible type "**Dict[str, object]"; expected "Optional[Validator]"  [arg-type]
great_expectations/agent/actions/run_onboarding_data_assistant.py:83: error: Argument 1 to "add_checkpoint" of "CloudDataContext" has incompatible type "**Dict[str, object]"; expected "Optional[Checkpoint]"  [arg-type]
```
as far as i can tell, this is correct usage and works, but please correct me if i'm wrong!